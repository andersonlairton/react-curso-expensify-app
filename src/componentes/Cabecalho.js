import React from 'react';
import {BrowserRouter,Route,Switch,Link,NavLink} from 'react-router-dom';

const Cabecalho=()=>
(
    <div>
        <h1>Exepensivo App</h1>
        <NavLink to="/" activeClassName="is-active"/* chamando elementos da classe */ exact={true}>Home</NavLink>
        <NavLink to="/help" activeClassName="is-active">Ajuda</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Editar</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create page</NavLink>
    </div>
);

export default Cabecalho;