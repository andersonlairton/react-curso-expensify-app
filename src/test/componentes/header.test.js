//import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import React from 'react';
import Cabecalho from '../../componentes/Cabecalho';
import toJSON from 'enzyme-to-json';

test('teste componente cabecalho', () => {
    const wrapper = shallow(<Cabecalho/>);
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find('h1').text()).toBe('Exepensivo App');
    /*
    const renderer = new ReactShallowRenderer();
    renderer.render(<Cabecalho/>);
    //console.log(renderer.getRenderOutput());
    expect(renderer.getRenderOutput()).toMatchSnapshot();
     */
});