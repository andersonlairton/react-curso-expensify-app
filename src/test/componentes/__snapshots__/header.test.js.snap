// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste componente cabecalho 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cabecalho />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Exepensivo App
</h1>,
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          exact={true}
          to="/"
>
          Home
</NavLink>,
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          to="/help"
>
          Ajuda
</NavLink>,
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          to="/edit"
>
          Editar
</NavLink>,
        <NavLink
          activeClassName="is-active"
          ariaCurrent="true"
          to="/create"
>
          Create page
</NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Exepensivo App",
        },
        "ref": null,
        "rendered": "Exepensivo App",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "ariaCurrent": "true",
          "children": "Home",
          "exact": true,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "ariaCurrent": "true",
          "children": "Ajuda",
          "to": "/help",
        },
        "ref": null,
        "rendered": "Ajuda",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "ariaCurrent": "true",
          "children": "Editar",
          "to": "/edit",
        },
        "ref": null,
        "rendered": "Editar",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "is-active",
          "ariaCurrent": "true",
          "children": "Create page",
          "to": "/create",
        },
        "ref": null,
        "rendered": "Create page",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Exepensivo App
</h1>,
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            exact={true}
            to="/"
>
            Home
</NavLink>,
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="/help"
>
            Ajuda
</NavLink>,
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="/edit"
>
            Editar
</NavLink>,
          <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="/create"
>
            Create page
</NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Exepensivo App",
          },
          "ref": null,
          "rendered": "Exepensivo App",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Home",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Ajuda",
            "to": "/help",
          },
          "ref": null,
          "rendered": "Ajuda",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Editar",
            "to": "/edit",
          },
          "ref": null,
          "rendered": "Editar",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Create page",
            "to": "/create",
          },
          "ref": null,
          "rendered": "Create page",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
